### ISSUE TITLE
change event not triggering when same value present in drop down / with remote data

### ISSUE THREAD

**Issue by cbichis** [select2/select2#4873]
Hi,

I am facing a problem with the change even not triggered when using multiple options with the same value (but diff label). According to https://github.com/select2/select2/issues/2796  this is fixed.
However, I am using remote data and this doesnt seems to be true.
Sample:

```
[
{id: 555000, text: "555000 - City 1"},
{id: 777123, text: "City 2"},
{id: 555000, text: "555000 - City 3"},
............
]
```

If I search for 555 I am getting as list of options first and the third, let's say I am selecting the first one. If I am searching again for 555 and selecting the third option the change it doesnt trigger!
But searching and selecting a diff id value it tiggers the change event...

So basically for remote data the change event is not fired if the old and the new ID's are same...

**Comment by alexweissman** [comment]
This could be related to the problem described in #4842.

**Comment by alexweissman** [comment]
This is definitely a duplicate of #4842.
